### PR TITLE
Shipment tracking

### PR DESCRIPTION
**Pull request by GillesWS**

### CODE REVIEW

**Comment by ThomasFrans** [comment]
Format please.

**Comment by ThomasFrans** [comment]
Could you just merge main and format again. I added a formatting fix last week which is pretty important (not having it is causing a lot of functions to now be incorrectly indented again, making reviewing this way harder). The commit with that fix isn't on your branch yet.

**Comment by GillesWS** [comment]
Like this?

File: package-lock.json
Comment (ThomasFrans) [inline package-lock.json:1]: https://github.com/Practice-Enterprise2/AD#pull-request-requirements

File: resources/css/app.css [hunk 1]
```diff
@@ -2,8 +2,49 @@
...
+    text-align: center;
+}
+.bg-lime {
+    background-color: lime;
+    color: black;
+    border-width: 2px;
+    border-radius: 6px;
+    text-align: center;
+    /* add the one's below to your own code. */
+}
```
Comment (ThomasFrans) [inline resources/css/app.css:47]: I don't really know where these are used but this is a bad idea as literally everyone will get these in their page like this. This change affects every page of our website. Don't add styles here except as sane overrides (like the one below).

File: resources/js/app.js [hunk 2]
```diff
@@ -1,2 +1,2 @@
 import './bootstrap';
-import * as te from 'tw-elements';
+//import * as te from 'tw-elements';
```
Comment (ThomasFrans) [inline resources/js/app.js:2]: Why is this removed? Just like with the styles, people probably depend on this. If this is wrong, it should be removed of course, but I don't know if it is wrong.

File: resources/views/graphs.blade.php [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+          $shipments[] = $data['sales'];
+      }
+    @endphp
+  </div>
+  <script type="text/javascript"
+    src="https://cdn.jsdelivr.net/npm/chart.js/dist/chart.umd.min.js"></script>
+  <script>
+    // setup 
+    const data = {
+      labels: <?php echo json_encode($day); ?>,
```
Comment (ThomasFrans) [inline resources/views/graphs.blade.php:44]: This is the same problem as with the other raw PHP tags: I don't think raw PHP is supported in Blade templates. Use `@php ... @endphp` like above.

File: resources/views/graphs.blade.php [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      datasets: [{
+        label: 'Weekly shipments per day',
+        data: <?php echo json_encode($shipments); ?>,
```
Comment (ThomasFrans) [inline resources/views/graphs.blade.php:47]: This is the same problem as with the other raw PHP tags: I don't think raw PHP is supported in Blade templates.

File: resources/views/layouts/navigation.blade.php [hunk 5]
```diff
@@ -75,9 +75,16 @@ class="mt-4 inline-flex items-center rounded-md border border-transparent border
             <x-dropdown-link :href="route('shipments.requests')">
               {{ __('Evaluate Shipment Requests') }}
             </x-dropdown-link>
-            <x-dropdown-link :href="route('shipments.index')">
-              {{ __('Show Confirmed Shipments') }}
+
+            <x-dropdown-link href="/shipments">
+              {{ __('Show Shipments') }}
```
Comment (ThomasFrans) [inline resources/views/layouts/navigation.blade.php:80]: Using named routes is better, so it's best not to change them for normal ones (as they break when the URL changes).

File: resources/views/layouts/navigation.blade.php [hunk 6]
```diff
@@ -75,9 +75,16 @@ class="mt-4 inline-flex items-center rounded-md border border-transparent border
...
             </x-dropdown-link>
+            @if (Auth::user()->roles()->first()->name == 'admin' ||
+                    Auth::user()->roles()->first()->name == 'employee')
```
Comment (ThomasFrans) [inline resources/views/layouts/navigation.blade.php:83]: Use permissions instead (except for the admin). There is a document in the `/docs/` directory that explains how to use permissions.

File: tailwind.config.js [hunk 7]
```diff
@@ -1,4 +1,6 @@
 /** @type {import('tailwindcss').Config} */
+
+// const twPlugin = require('tw-elements/dist/plugin');
```
Comment (ThomasFrans) [inline tailwind.config.js:3]: https://github.com/Practice-Enterprise2/AD#pull-request-requirements

File: resources/views/shipments.blade.php [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+                      @elseif ($shipment->status == 'Canceled')
+                        <div
+                          class="rounded-md border-2 border-red-900 bg-red-500 py-2 text-center text-base font-bold text-white">
+                          Canceled</div>
+                      @endif
+
+                    </td>
+                    <td
+                      class="whitespace-no-wrap border-b border-gray-400 px-6 py-4">
+                      {{-- <a href="{{ route('shipments.showShipments_details', $shipment->id) }}"><button type="button" class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 mr-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800">Details</button></a> --}}
```
Comment (ThomasFrans) [inline resources/views/shipments.blade.php:214]: https://github.com/Practice-Enterprise2/AD#pull-request-requirements

File: resources/views/shipments.blade.php [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+                class="border-b border-gray-400 px-6 py-3 text-left text-sm font-bold uppercase text-gray-600"
+                id="status">@sortablelink('status', 'ShipmentStatus')</th>
+              <th
+                class="border-b border-gray-400 px-6 py-3 text-left text-sm font-bold uppercase text-gray-600">
+              </th>
+            </tr>
+          </thead>
+          <tbody>
+            @if (Auth::user()->roles()->first()->name == 'admin' ||
+                    Auth::user()->roles()->first()->name == 'employee')
```
Comment (ThomasFrans) [inline resources/views/shipments.blade.php:64]: Use permissions instead.

File: resources/views/shipments.blade.php [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+                    @elseif ($shipment->status == 'Canceled')
+                      <div
+                        class="rounded-md border-2 border-red-900 bg-red-500 py-2 text-center text-base font-bold text-white">
+                        Canceled</div>
+                    @endif
+
+                  </td>
+                  <td
+                    class="whitespace-no-wrap border-b border-gray-400 px-6 py-4">
+                    {{--                                 <a href="{{ route('shipments.showShipments_details', $shipment->id) }}"><button type="button" class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 mr-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800">Details</button></a> --}}
```
Comment (ThomasFrans) [inline resources/views/shipments.blade.php:116]: https://github.com/Practice-Enterprise2/AD#pull-request-requirements

File: resources/views/shipments.blade.php [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+                      cancelCancelButton.addEventListener('click', function() {
+                        modal.classList.add('hidden');
+                      });
+                    </script>
+                  </td>
+                </tr>
+              @endforeach
+            @else
+              <!-- Users can only see their own shipment -->
+              <!-- -->
```
Comment (ThomasFrans) [inline resources/views/shipments.blade.php:161]: https://github.com/Practice-Enterprise2/AD#pull-request-requirements

File: resources/views/show.blade.php
Comment (ThomasFrans) [inline resources/views/show.blade.php:1]: Remove this file if it isn't used.

File: routes/web.php
Comment (ThomasFrans) [inline routes/web.php:1]: Restore this file back to how it was and only add your routes. Don't just remove the middleware groups please.

**Review by ThomasFrans — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (GillesWS) [reply package-lock.json:1]: Someone else updated this i think

**Comment by ThomasFrans** [comment]
Format please

**Comment by ThomasFrans** [comment]
This contains many mistakes and will break the main branch, including:

- Git merge errors
- Duplicate code

It will have to go into the other main branch and be demonstrated separately, if agreed by @jentevs and @inancalp.

**Pull request closed without merging**